src/main/java/com/arpnetworking/tsdcore/model/AggregationMessage.java

```diff
@@ -174,17 +209,26 @@ public int getLength() {
         return _length;
     }
 
-    private AggregationMessage(final GeneratedMessageV3 message) {
-        this(message, serialize(message).length());
+    public AggregationMode getAggregationMode() {
+        return _aggregationMode;
+    }
+
+    private AggregationMessage(final GeneratedMessageV3 message, final AggregationMode aggregationMode) {
+        this(message, serialize(message).length(), aggregationMode);
     }
 
-    private AggregationMessage(final GeneratedMessageV3 message, final int length) {
+    private AggregationMessage(
+            final GeneratedMessageV3 message,
+            final int length,
+            final AggregationMode aggregationMode) {
         _length = length;
         _message = message;
+        _aggregationMode = aggregationMode;
     }
 
     private final GeneratedMessageV3 _message;
     private final int _length;
+    private final AggregationMode _aggregationMode;
```

Comment:
I'm against putting the aggregation mode on the data itself.  This is a routing problem and has nothing to do with the data.

Reply:
Fixed.